include/dxc/WinAdapter.h

```diff
@@ -912,32 +912,45 @@ unsigned int SysStringLen(const BSTR bstrString);
 #define CP_ACP 0
 #define CP_UTF8 65001 // UTF-8 translation.
 
-// Convert Windows codepage value to locale string
-const char *CPToLocale(uint32_t CodePage);
+// RAII style mechanism for setting/unsetting a locale for the specified Windows
+// codepage
+class ScopedLocale {
+  const char *m_locale;
+
+public:
+  explicit ScopedLocale(uint32_t codePage)
+      : m_locale(SetLocaleForCodePage(LC_ALL, codePage)) {}
+  ~ScopedLocale() {
+    if (m_locale != nullptr) {
+      setlocale(LC_ALL, m_locale);
+    }
+  }
+
+private:
+  // Sets a locale for the specified Windows codepage
+  const char *SetLocaleForCodePage(int Category, uint32_t CodePage) {
+    if (CodePage != CP_UTF8) {
+      assert(false && "Support for Linux only handles UTF8 code pages");
+    }
```

Comment:
Why isn't this just:
```suggestion
    assert(CodePage == CP_UTF8 && "Support for Linux only handles UTF8 code pages");
```

Reply:
It could be.  Just a style choice I made that the time.  I am going to leave it as-is for now.

Reply:
Please don't - `assert(false)` is an antipattern. It's better to assert the condition directly.